### PR TITLE
Backend: Add endpoint for exporting account statement as CSV file

### PR DESCRIPTION
**Pull request by bisgardo**
## Purpose

Provide a solution for users wanting to export a full account statement. No support for filtering is implemented. It could make sense to add optional time bounds.

Part of epic "Export transactions from CCDScan".

Most accounts will process in less than 1s. Account `35CJPZohio6Ztii2zy1AYzJKvuxbGG44wrBn7hLHiYLoF2nxnh` which as of this writing has 4339327 statement entries takes 5-6 secs locally and results in a 200MB download (and has too many entries for spreadsheets to open it). This could be used maliciously, so adding some form of rate limiting may be desirable.

## Changes

Add endpoint `/rest/export/statement?accountAddress=<account-address>` to the backend that responds with a CSV file `statement-<account-address>.csv` containing all records formatted like:

```csv
Time,Amount (CCD),Label
2022-10-07 08:00:19Z,15526.040342,FinalizationReward
2022-10-08 08:00:01Z,38.824926,TransactionFeeReward
2022-10-08 08:00:01Z,256225.95173,BakerReward
2022-10-08 08:00:01Z,15523.962755,FinalizationReward
...
```

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.

### CODE REVIEW

File: backend/Application/Api/Rest/ExportController.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+        var query = dbContext.AccountStatementEntries
+            .AsNoTracking()
+            .Where(x => x.AccountId == account.Id);
+
+        var scalarQuery = query.Select(x => new
+        {
+            x.Timestamp,
+            x.EntryType,
+            x.Amount,
```
Comment (shjortConcordium) [inline backend/Application/Api/Rest/ExportController.cs:48]: Is this amount the "balance change" for the given account?

File: backend/Application/Api/Rest/ExportController.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        });
+        var values = await scalarQuery.ToListAsync();
+        // Should use something like 'CsvHelper' (see 'https://joshclose.github.io/CsvHelper/examples/writing/write-anonymous-type-objects/')?
```
Comment (shjortConcordium) [inline backend/Application/Api/Rest/ExportController.cs:51]: Make this a TODO, do it or remove it? :thinking:

Comment (bisgardo) [reply backend/Application/Api/Rest/ExportController.cs:51]:
```suggestion
        // TODO Should use something like 'CsvHelper' (see 'https://joshclose.github.io/CsvHelper/examples/writing/write-anonymous-type-objects/')?
```

Was thinking to get this merged first as it works. Then try it out and see how it affects performance.

File: backend/Application/Api/Rest/ExportController.cs (hunk 1 above)
Comment (bisgardo) [reply backend/Application/Api/Rest/ExportController.cs:48]: yes

**Comment by bisgardo** [comment]
> It looks okay to me. I haven't tested it though, I assume that is very cumbersome to do?

It's quite cumbersome, yes. I can help you set it up but it would be easier if we just deploy it to stagenet and test it from there.

**Pull request merged**